tests/phpunit/tests/st/strategy/shortcode/test-wpml-pb-shortcode-content-wrapper.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * @group page-builders-shortcodes
+ * @group wpmltm-2970
+ */
+class Test_WPML_PB_Shortcode_Content_Wrapper extends WPML_PB_TestCase {
+
+	/**
+	 * @test
+	 * @dataProvider dp_get_wrapped_content
+	 *
+	 * @param string $content
+	 * @param string $expected_content
+	 */
+	public function it_should_get_wrapped_content( $content, $expected_content ) {
+		$valid_shortcodes = array(
+			'av_button',
+			'av_notification',
+			'shortcode_nested',
+			'shortcode_A',
+		);
+
+		$subject = new WPML_PB_Shortcode_Content_Wrapper( $content, $valid_shortcodes );
+		$this->assertEquals( $expected_content, $subject->get_wrapped_content() );
+	}
+
+	public function dp_get_wrapped_content() {
+		return array(
+			'basic text'                                       => $this->get_fixture_contents( 'basic-text' ),
+			'with open and closing shortcode'                  => $this->get_fixture_contents( 'with-open-and-closing-shortcode' ),
+			'with 2 shortcode side by side'                    => $this->get_fixture_contents( 'with-2-shortcodes-side-by-side' ),
+			'with nested shortcodes'                           => $this->get_fixture_contents( 'with-nested-shortcodes' ),
+			'start and end with shortcodes'                    => $this->get_fixture_contents( 'start-and-end-with-shortcodes' ),
+			'"empty" strings between 2 shortcodes (no change)' => $this->get_fixture_contents( 'empty-strings-between-2-shortcodes' ),
+			'with invalid shortcodes (should wrap it)'         => $this->get_fixture_contents( 'with-invalid-shortcodes' ),
+			'with brackets in shortcode attributes'            => $this->get_fixture_contents( 'with-brackets-in-shortcode-attributes' ),
+			'with multibyte chars'                             => $this->get_fixture_contents( 'with-multibyte-chars' ),
+		);
```

Comment:
@strategio Why not just `glob` the directory to find the files. Then it's easier to add more cases by just adding new `xxx.html` file and `xxx-wrapped.html` without having to adjust the php.